src/index.js

```diff
@@ -55,44 +70,47 @@ const applyCommit = (state, commitId, reducer) => {
       meta: Object.assign({}, actionToCommit[1].meta,
         {optimistic: null})
     });
-    return state.set('history', state.get('history').set(actionToCommit[0], newAction))
+    return {
+      ...state,
+      history: [newAction].concat(state.history.slice(1))
+    };
   }
 };
 
 const applyRevert = (state, revertId, reducer) => {
-  const history = state.get('history');
-  const beforeState = state.get('beforeState');
+  const { beforeState, history } = state;
   let newHistory;
   // If the action to revert is the first in the queue (most common scenario)
-  if (history.first().meta.optimistic.id === revertId) {
-    const historyWithoutRevert = history.shift();
-    const nextOptimisticIndex = historyWithoutRevert.findIndex(action => action.meta && action.meta.optimistic && !action.meta.optimistic.isNotOptimistic && action.meta.optimistic.id);
+  if (history[0].meta.optimistic.id === revertId) {
+    const historyWithoutRevert = shift(history);
```

Comment:
Actually, I just realised I can probably just replace this with `history.slice(1)` for the same effect.